bodo/pandas/plan.py

```diff
@@ -695,6 +758,30 @@ def _exec_plan(plan):
             print("profile_time execute", time.perf_counter() - start_time)
         return ret
 
+    # Find duplicated subqueries in the given plan and materialize them.
+    while True:
+        # Find the next duplicated subquery in the given plan.
+        top_most_subquery = plan.get_subqueries()
+        if top_most_subquery is not None:
+            # If there is one then go back from the plan to the
+            # dataframe or series object.
+            materialized_frame_or_series = wrap_plan(top_most_subquery)
```

Comment:
I tried another approach where the plan points back to the dataframe and this line then follows that back link.  That approach would have the advantage that if there was another query that needed that dataframe that it wouldn't be computed again.  The way this PR is now, it will only temporarily materialize within the context of one plan submitted to execute_plan.  The approach with the back pointer fails though in parallel mode with that error about spawn mode already running seemingly on the line where the back pointer is set.  I don't understand that at all.  If there are thoughts of a solution then we could consider turning this on.